src/main/java/frc/robot/subsystems/BallPath/Elevator/ElevatorImpl.java

```diff
@@ -80,55 +83,93 @@ public void task() throws Exception {
             case PRIME:
-                if (ballPresent) {
+                // checking for ball
+                detectedColor = elevatorColorSensor.getColor();
+                SmartDashboard.putNumber("Red", detectedColor.red);
+                SmartDashboard.putNumber("Green", detectedColor.green);
+                SmartDashboard.putNumber("Blue", detectedColor.blue);
+                // System.out.println("Elevator: Priming");
+                if(detectedColor.red > 0.31){
+                    detectedColorElevator = 1;
+                    // System.out.println("RED");
+                }else if(detectedColor.blue > 0.31){
+                    detectedColorElevator = 2;
+                    // System.out.println("BLUE");
+                }else{
+                    detectedColorElevator = 0;
+                }
+
+                if (detectedColorElevator == 1 || detectedColorElevator == 2){
+                    ballPresent = true;
                     this.elevator.set(MOTOR_SPEED);
-                } else {
-                    this.elevator.stopMotor();
+                    // System.out.println("Elevator: Ball present is true");
+                }else{
+                    // System.out.println("Elevator: Ball present is false\n@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@\n@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@\n@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@\n@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@");
+                    ballPresent = false;
+                    this.elevator.set(0);
                 }
                 break;
             case IN:
                 if (!shooter.blocking()) {
                     this.elevator.set(MOTOR_SPEED);
+                    ballPresent = false;
                 }
                 break;
             case OUT:
                 this.elevator.set(-MOTOR_SPEED);
+                ballPresent = false;
                 break;
             case RUN:
                 this.elevator.set(MOTOR_SPEED);
+                ballPresent = false;
                 break;
             case INDEX:
-                Color detectedColor = elevatorColorSensor.getColor();
-                SmartDashboard.putNumber("Red", detectedColor.red);
-                SmartDashboard.putNumber("Green", detectedColor.green);
-                SmartDashboard.putNumber("Blue", detectedColor.blue);
-                System.out.println("Elevator Indexing");
-                // boolean stateChanged = ballPresent != lastPresent;
-                if(detectedColor.red > 0.33){
-                    detectedColorElevator = 1;
-                    System.out.println("RED");
-                }else if(detectedColor.blue > 0.32){
-                    detectedColorElevator = 2;
-                    System.out.println("BLUE");
-                }else{
-                    detectedColorElevator = 0;
+                count ++;
+                // System.out.println(count + " " + (count % 10));
+                if (!ballPresent || (count % 10) == 0){
```

Comment:
What's up with `count % 10 == 0`? So every 10th iteration of looping while the action is `INDEX`, we do this? Is this meant to just simulate having a slower poll rate for this specific action?

Reply:
I think Massimo added that in, but I believe its purpose was that we still check if there is a ball even if we know there is a ball, but at a 1/10th polling rate?

Reply:
Okay, that's what it seemed like. Makes enough sense, though this does indicate that it should probably be extracted out to its own separate subsystem with its own slower polling rate.